### PR TITLE
More use of the IsStandardModel and IsValidModel functions

### PR DESCRIPTION
**Pull request by Moris-Onz**
Try to use IsStandardModel and IsValidModel more often instead of prescribing limits every time

### CODE REVIEW

**Review by Pirulax — APPROVED** [review]

File: Client/mods/deathmatch/logic/CClientModelCacheManager.cpp [hunk 1]
```diff
@@ -385,10 +385,10 @@ void CClientModelCacheManagerImpl::ProcessVehicleList(std::map<ushort, float>& o
     const ulong ulTimeNow = CClientTime::GetTime();
     for (std::vector<CClientVehicle*>::const_iterator iter = vehicleList.begin(); iter != vehicleList.end(); ++iter)
     {
-        CClientVehicle* pVehicle = *iter;
-        const ushort    usModelId = pVehicle->GetModel();
+        CClientVehicle*      pVehicle = *iter;
+        const unsigned short usModelId = pVehicle->GetModel();
 
-        if (usModelId < 400 || usModelId > 611)
+        if (CClientVehicleManager::IsStandardModel(usModelId))
```
Comment (Pirulax) [inline Client/mods/deathmatch/logic/CClientModelCacheManager.cpp:391]:
```suggestion
        if (!CClientVehicleManager::IsStandardModel(usModelId))
```

File: Client/mods/deathmatch/logic/CClientVehicle.cpp [hunk 2]
```diff
@@ -1052,10 +1052,10 @@ void CClientVehicle::SetModelBlocking(unsigned short usModel, unsigned char ucVa
         m_ucMaxPassengers = CClientVehicleManager::GetMaxPassengerCount(usModel);
 
         // Reset handling to fit the vehicle
-        if (IsLocalEntity() || !(usModel < 400 || usModel > 611))
+        if (IsLocalEntity() || !CClientVehicleManager::IsStandardModel(usModel))
         {
-            ushort usHandlingModelID = usModel;
-            if (usHandlingModelID < 400 || usHandlingModelID > 611)
+            unsigned short usHandlingModelID = usModel;
+            if (CClientVehicleManager::IsStandardModel(usHandlingModelID))
```
Comment (Pirulax) [inline Client/mods/deathmatch/logic/CClientVehicle.cpp:1058]:
```suggestion
            if (!CClientVehicleManager::IsStandardModel(usHandlingModelID))
```

File: Client/mods/deathmatch/logic/CClientVehicleManager.cpp [hunk 3]
```diff
@@ -472,7 +472,7 @@ eClientVehicleType CClientVehicleManager::GetVehicleType(unsigned long ulModel)
 unsigned char CClientVehicleManager::GetMaxPassengerCount(unsigned long ulModel)
 {
     // Use parent model ID for non-standard vehicle model IDs.
-    if ((ulModel < 400 || ulModel > 611) && IsValidModel(ulModel))
+    if (IsStandardModel(ulModel) && IsValidModel(ulModel))
```
Comment (Pirulax) [inline Client/mods/deathmatch/logic/CClientVehicleManager.cpp:475]:
```suggestion
    if (!IsStandardModel(ulModel) && IsValidModel(ulModel))
```

File: Client/mods/deathmatch/logic/CClientVehicle.cpp [hunk 4]
```diff
@@ -1052,10 +1052,10 @@ void CClientVehicle::SetModelBlocking(unsigned short usModel, unsigned char ucVa
         m_ucMaxPassengers = CClientVehicleManager::GetMaxPassengerCount(usModel);
 
         // Reset handling to fit the vehicle
-        if (IsLocalEntity() || !(usModel < 400 || usModel > 611))
+        if (IsLocalEntity() || !CClientVehicleManager::IsStandardModel(usModel))
```
Comment (Pirulax) [inline Client/mods/deathmatch/logic/CClientVehicle.cpp:1055]:
```suggestion
        if (IsLocalEntity() || CClientVehicleManager::IsStandardModel(usModel))
```

File: Client/mods/deathmatch/logic/CNetAPI.cpp [hunk 5]
```diff
@@ -1563,7 +1563,7 @@ void CNetAPI::WriteVehiclePuresync(CClientPed* pPlayerModel, CClientVehicle* pVe
     // Use parent model ID for non-standard vehicle model IDs.
     // This avoids a mismatch between client and server, ensuring doors and damage sync correctly.
     int iModelID = pVehicle->GetModel();
-    if (iModelID < 400 || iModelID > 611)
+    if (CClientVehicleManager::IsStandardModel(iModelID))
```
Comment (Pirulax) [inline Client/mods/deathmatch/logic/CNetAPI.cpp:1566]:
```suggestion
    if (!CClientVehicleManager::IsStandardModel(iModelID))
```

File: Client/mods/deathmatch/logic/CVehicleNames.cpp [hunk 6]
```diff
@@ -288,11 +283,11 @@ unsigned int CVehicleNames::GetVehicleModel(const char* szName)
 const char* CVehicleNames::GetVehicleTypeName(unsigned long ulModel)
 {
     // Use parent model ID for non-standard vehicle model IDs.
-    if ((ulModel < 400 || ulModel > 611) && CClientVehicleManager::IsValidModel(ulModel))
+    if (CClientVehicleManager::IsStandardModel(ulModel) && CClientVehicleManager::IsValidModel(ulModel))
```
Comment (Pirulax) [inline Client/mods/deathmatch/logic/CVehicleNames.cpp:286]: Missing negation

**Comment by botder** [comment]
Why were the suggestions by Pirulax marked as resolved without any changes to the code?

Since you started refactoring/removing the redundant logic, why not fix the naming of the functions?
There still is a mix of IsValidModel and IsStandardModel and both do not make it clear what kind of models are acceptable.

**Pull request closed without merging**